x/tier/keeper/query_get_users_pool_data.go

```diff
@@ -0,0 +1,124 @@
+package keeper
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ammtypes "github.com/elys-network/elys/x/amm/types"
+	assetprofiletypes "github.com/elys-network/elys/x/assetprofile/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+	"github.com/elys-network/elys/x/tier/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type detailPool struct {
+	Title string
+	Pool  ammtypes.Pool
+}
+
+func (k Keeper) GetUsersPoolData(goCtx context.Context, req *types.QueryGetUsersPoolDataRequest) (*types.QueryGetUsersPoolDataResponse, error) {
+
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	usdcDenom, found := k.assetProfileKeeper.GetUsdcDenom(ctx)
+
+	if !found {
+		return nil, errors.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+	}
+
+	listCommitments, pagination, err := k.commitement.GetAllCommitmentsWithPagination(ctx, req.Pagination)
+
+	if err != nil {
+		return nil, err
+	}
+
+	tokenPrice := k.oracleKeeper.GetAssetPriceFromDenom(ctx, usdcDenom)
+	params := k.stablestakeKeeper.GetParams(ctx)
+
+	usersData := []*types.UserData{}
+
+	pools := map[uint64]detailPool{}
+
+	for _, user := range listCommitments {
+
+		u := types.UserData{
+			User:  user.Creator,
+			Pools: []*types.Pool{},
+		}
+
+		for _, commitment := range user.CommittedTokens {
+			if strings.HasPrefix(commitment.Denom, "stablestake/share") {
+				fiatValue := commitment.Amount.ToLegacyDec().Mul(params.RedemptionRate).Mul(tokenPrice)
```

Comment:
@fenriz07 what are the expected denom values for `commitment.Denom`? if it anything other than `USDC` then the tokenPrice should be retrieved using the `commitment.Denom`:

```
	tokenPrice := k.oracleKeeper.GetAssetPriceFromDenom(ctx, commitment.Denom)
```

right?

Reply:
all good token price is actually used to covent pool values from usdc to usd